File: Sql_Easy.vb
Public Class SqlEasy
    ' SqlEasyクラスは、SQL Serverに接続してSQL文を実行するためのクラスです。
    ' 使い方は、Sqlクラスのインスタンスを作成し、Executeメソッドを呼び出すだけです。 

    ' 初期化は接続プロパティを設定するだけでOKです。
    ' また、本文には列の配列を用意しておいてください。引数として使用します。

    ' 例1:DBへ入力する場合
    ' sql.SetDB("INSERT INTO table_name (column1, column2) VALUES (value1, value2)")
    ' 例2:DBから取得する場合
    ' Dim dt As DataTable = sql.GetDB("SELECT column1, column2 FROM table_name", New String() {"column1", "column2"})

    '＜ちょっとした小ネタ＞このまま使用できますが、この関数の末尾に文字列としてエラーコードを自分で記述すると、失敗時にその文章を表示します。
    
    ' 接続用の文字列
    Private Data_Source As String
    Private User_ID As String
    Private Password As String
    Private Inital_Catalog As String

    Sub New(Sourse As String,
            ID As String,
            Password As String,
            Database As String)
        Me.Data_Source = Sourse
        Me.User_ID = ID
        Me.Password = Password
        Me.Inital_Catalog = Database
    End Sub

    ' SQL文を実行するメソッド(set)
    Public Function SetDB(Sql As String) As Boolean
        Try
            Dim cn As New Microsoft.Data.SqlClient.SqlConnection
            cn.ConnectionString = "Data Source=" & Me.Data_Source & ";Initial Catalog=" & Me.Inital_Catalog & ";User ID=" & Me.User_ID & ";Password=" & Me.Password & ";"

            cn.Open()

            Dim cmd As New Microsoft.Data.SqlClient.SqlCommand
            cmd.CommandText = Sql
            cmd.Connection = cn
            cmd.ExecuteNonQuery()

            cn.Close()
            cn.Dispose()
            Return True
        Catch ex As Exception
            MsgBox(ex.Message)
            Return False
        End Try
    End Function

    ' SQL文を実行するメソッド(get)
    Public Function GetDB(Sql As String,
            Columns As String()) As DataTable
        Try

            Dim cn As New Microsoft.Data.SqlClient.SqlConnection
            cn.ConnectionString = "Data Source=" & Me.Data_Source & ";Initial Catalog=" & Me.Inital_Catalog & ";User ID=" & Me.User_ID & ";Password=" & Me.Password & ";"

            cn.Open()

            Dim cmd As New Microsoft.Data.SqlClient.SqlCommand
            cmd.CommandText = Sql
            cmd.Connection = cn

            Dim dr As Microsoft.Data.SqlClient.SqlDataReader = cmd.ExecuteReader()

            Dim dtData As New DataTable
            For i As Integer = 0 To Columns.Length - 1
                dtData.Columns.Add(Columns(i))
            Next i

            While dr.Read()
                Dim drData As DataRow = dtData.NewRow()
                For i As Integer = 0 To Columns.Length - 1
                    drData(i) = dr(Columns(i))
                Next i
                dtData.Rows.Add(drData)
            End While

            dr.Close()
            cn.Close()
            cn.Dispose()

            Return dtData
        Catch ex As Exception
            MsgBox(ex.Message)
            Return Nothing
        End Try
    End Function

    Function SetDB(Sql As String, ErrorText As String) As Boolean
        Try
            Dim cn As New Microsoft.Data.SqlClient.SqlConnection
            cn.ConnectionString = "Data Source=" & Me.Data_Source & " Initial Catalog=" & Me.Inital_Catalog & " User ID=" & Me.User_ID & " Password=" & Me.Password & " "

            cn.Open()

            Dim cmd As New Microsoft.Data.SqlClient.SqlCommand
            cmd.CommandText = Sql
            cmd.Connection = cn
            cmd.ExecuteNonQuery()

            cn.Close()
            cn.Dispose()
            Return True
        Catch ex As Exception
            MsgBox(ErrorText)
            Return False
        End Try
    End Function

    ' SQL文を実行するメソッド(get)
    Public Function GetDB(Sql As String,
            Columns As String(),
            ErrorText As String) As DataTable
        Try

            Dim cn As New Microsoft.Data.SqlClient.SqlConnection
            cn.ConnectionString = "Data Source=" & Me.Data_Source & " Initial Catalog=" & Me.Inital_Catalog & " User ID=" & Me.User_ID & " Password=" & Me.Password & " "

            cn.Open()

            Dim cmd As New Microsoft.Data.SqlClient.SqlCommand
            cmd.CommandText = Sql
            cmd.Connection = cn

            Dim dr As Microsoft.Data.SqlClient.SqlDataReader = cmd.ExecuteReader()

            Dim dtData As New DataTable
            For i As Integer = 0 To Columns.Length - 1
                dtData.Columns.Add(Columns(i))
            Next i

            While dr.Read()
                Dim drData As DataRow = dtData.NewRow()
                For i As Integer = 0 To Columns.Length - 1
                    drData(i) = dr(Columns(i))
                Next i
                dtData.Rows.Add(drData)
            End While

            dr.Close()
            cn.Close()
            cn.Dispose()

            Return dtData
        Catch ex As Exception
            MsgBox(ErrorText)
            Return Nothing
        End Try
    End Function
End Class

File: Sql_Easy.cs
namespace Sql_Easy
{
    using System;
    using System.Collections.Generic;
    using System.Data;
    using System.Data.SqlClient;

    public class Sql_Easy
    {
        private string contectString;
        private List<string> selectColumn;

        public Sql_Easy(string Source, string Id, string Pass, string DataBase)
        {
            this.contectString = "Data Source=" + Source + ";Initial Catalog=" + DataBase + ";User ID=" + Id + ";Password=" + Pass + ";";
            this.selectColumn = new List<string>();
        }

        public DataTable? SqlCmd
            (
            string sqlType,
            string TableName,
            List<string> Column,
            string eMessage ="default",
            List<string>? newValue = default,
            List<string>? whereColumn=default,
            List<string>? whereValue = default
            )
        {
            try
            {
                DataTable table = new DataTable();
                using (SqlConnection connection = new SqlConnection(contectString))
                {
                    connection.Open();

                    string sql = "";
                    if (sqlType == "SELECT")
                    {
                        sql += "SELECT ";
                        if (Column == null)
                        {
                            sql += "* FROM " + TableName + "";
                        }
                        else
                        {
                            for (int i = 0; i < Column.Count; i++)
                            {
                                sql += Column[i] + ", ";
                            }
                            sql += " FROM " + TableName + "";
                        }
                    }
                    else if (sqlType == "INSERT") { 
                        sql += "INSERT INTO " + TableName + "(";
                        for (int i = 0; i < Column.Count; i++)
                        {
                            sql += Column[i];
                            if (i != Column.Count - 1)
                            {
                                sql += ", ";
                            }
                        }
                        sql += ") VALUES (";
                        for (int i = 0; i < Column.Count; i++)
                        {
                            sql += "@" + i.ToString() + "Value";
                            if (i != Column.Count - 1)
                            {
                                sql += ", ";
                            }
                        }
                        sql += ")";
                    }
                    else if (sqlType == "UPDATE")
                    {
                        sql += "UPDATE " + TableName + " SET ";
                        for (int i = 0; i < Column.Count; i++)
                        {
                            sql += Column[i] + " = @" + i.ToString() + "Value";
                            if (i != Column.Count - 1)
                            {
                                sql += ", ";
                            }
                        }
                    }else{
                        return null;
                    }
                    if (whereColumn != null)
                    {
                        sql += " WHERE ";
                        for (int i = 0; i < whereColumn.Count; i++)
                        {
                            sql += whereColumn[i] + " = @" + i.ToString() + "WhereValue";
                            if (i != whereColumn.Count - 1)
                            {
                                sql += " AND ";
                            }
                        }
                    }
                    sql += ";";

                    SqlCommand command = new SqlCommand(sql, connection);

                    if (newValue != null)
                    {
                        for (int i = 0; i < newValue.Count; i++)
                        {
                            command.Parameters.AddWithValue("@" + i.ToString() + "Value", newValue[i]);
                        }
                    }
                    if(whereValue != null)
                    {
                        for (int i = 0; i < whereValue.Count; i++)
                        {
                            command.Parameters.AddWithValue("@" + i.ToString() + "WhereValue", whereValue[i]);
                        }
                    }

                    if (sqlType == "SELECT")
                    {
                        SqlDataReader reader = command.ExecuteReader();
                        while (reader.Read())
                        {
                            DataRow row = table.NewRow();
                            foreach (string column in Column)
                            {
                                Type type = reader[column].GetType();
                                if (type.Name == "DateTime")
                                {
                                    DateTime time = reader.GetDateTime(reader.GetOrdinal(column));
                                    row[column] = time.ToString("yyyy/MM/dd HH:mm:ss");
                                }
                                else
                                {
                                    row[column] = Convert.ChangeType(reader[column].ToString(), type);
                                }
                            }
                            table.Rows.Add(row);
                        }
                    }
                    else if (sqlType == "INSERT" || sqlType == "UPDATE" || sqlType == "DELETE")
                    {
                        command.ExecuteNonQuery();
                        connection.Close();
                        table.Columns.Add("collected!");
                    }
                }
                return table;
            }
            catch (SqlException e)
            {
                if(eMessage == null)
                    Console.WriteLine("エラー: " + e.Message);
                else
                    Console.WriteLine(eMessage);
                return null;
            }
        }
    }
}
